pkgs/development/python-modules/diagrams/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, fetchPypi
+, isPy3k
+, graphviz
+}:
+
+buildPythonPackage rec {
+  pname = "diagrams";
+  version = "0.17.0";
+  disabled = ! isPy3k;
+
+  # TODO consider packaging from Github: https://github.com/mingrammer/diagrams
```

Comment:
This should be easy. Use `fetchFromGitHub` instead of `fetchPypi` and set `format = "pyproject";`. I assume we would then have the `tests/` directory.

Reply:
I spent a few hours working on this. I am far from a python expert, but it seems like the GitHub repo is structurally quite different from the pypi tarball. I will upload a new branch in a bit with my (unsuccessful) efforts.

Reply:
You can see my (stalled) progress on packaging from GitHub here: https://github.com/NixOS/nixpkgs/pull/102060

@evanjs also had what is probably a terrible idea for tests, which is to 'cherry-pick' just the tests from GitHub...